macaroonbakery/macaroon.py

```diff
@@ -0,0 +1,291 @@
+    '''Return the macaroon version that should be used
+    with the given bakery version.
+
+    @param bakery_version the bakery version
+    @return macaroon_version the derived macaroon version
+    '''
+    if bakery_version in [bakery.BAKERY_V0, bakery.BAKERY_V1]:
+        return MACAROON_V1
+    return MACAROON_V2
+
+
+def parse_local_location(loc):
+    '''Parse a local caveat location as generated by
+    LocalThirdPartyCaveat.
+
+    This is of the form:
+
+        local <version> <pubkey>
+
+    where <version> is the bakery version of the client that we're
+    adding the local caveat for.
+
+    It returns false if the location does not represent a local
+    caveat location.
+    @return a tuple of location and if the location is local.
+    '''
+    if not(loc.startswith('local ')):
+        return (), False
+    v = bakery.BAKERY_V1
+    fields = loc.splits()
+    fields = fields[1:]  # Skip 'local'
+    if len(fields) == 2:
+        try:
+            v = int(fields[0])
+        except ValueError:
+            return (), False
+        fields = fields[1:]
+    if len(fields) == 1:
+        return (base64.b64decode(fields[0]), v), True
+    return (), False
+
+
+class ThirdPartyLocator:
+    def __init__(self):
+        self._store = {}
+
+    def third_party_info(self, loc):
+        '''Return information on the third party at the given location.
+
+        It returns None if no match is found.
+
+        @param loc string
+        @return: string
+        '''
+        return self._store.get(loc)
+
+    def add_info(self, loc, info):
+        self._store[loc.rstrip('\\')] = info
+
+
+class ThirdPartyCaveatInfo:
+    '''ThirdPartyCaveatInfo holds the information decoded from
+    a third party caveat id.
+    '''
+    def __init__(self, condition, first_party_public_key, third_party_key_pair,
+                 root_key, caveat, version):
+        '''
+        @param condition holds the third party condition to be discharged.
+        This is the only field that most third party dischargers will
+        need to consider.
+        @param first_party_public_key 	holds the nacl public key of the party
+        that created the third party caveat.
+        @param third_party_key_pair holds the nacl private used to decrypt
+        the caveat - the key pair of the discharging service.
+        @param root_key bytes holds the secret root key encoded by the caveat.
+        @param caveat holds the full encoded base64 string caveat id from
+        which all the other fields are derived.
+        @param version holds the version that was used to encode
+        the caveat id.
+        '''
+        self.condition = condition,
+        self.first_party_public_key = first_party_public_key,
+        self.third_party_key_pair = third_party_key_pair,
+        self.root_key = root_key,
+        self.caveat = caveat,
+        self.version = version,
+
+    def __eq__(self, other):
+        if self.condition != other.condition:
```

Comment:
This can be converted in a single if statement:
```
return (
    self.condition == other.condition and
    self.first_party_public_key == other.first_party_public_key and
    ...
)